misk/src/main/kotlin/misk/web/actions/WebActionMetadataAction.kt

```diff
@@ -17,10 +24,52 @@ class WebActionMetadataAction : WebAction {
   @Get("/api/webactionmetadata")
   @RequestContentType(MediaTypes.APPLICATION_JSON)
   @ResponseContentType(MediaTypes.APPLICATION_JSON)
-  @Unauthenticated // TODO(adrw) add AccessAnnotation for AdminTab
+  @Unauthenticated // TODO(adrw) add AccessAnnotation for AdminDashboardTab
   fun getAll(): Response {
     return Response(webActionMetadata = servletProvider.get().webActionsMetadata)
   }
 
   data class Response(val webActionMetadata: List<WebActionMetadata>)
+}
+
+data class WebActionMetadata(
+  val name: String,
+  val function: String,
+  val functionAnnotations: List<String>,
+  val requestMediaTypes: List<String>,
+  val responseMediaType: String?,
+  val parameterTypes: List<String>,
+  val returnType: String,
+  val pathPattern: String,
+  val applicationInterceptors: List<String>,
+  val networkInterceptors: List<String>,
+  val dispatchMechanism: DispatchMechanism
+)
+
+internal fun WebActionMetadata(
```

Comment:
is it possible to enclose this in WebActionMetadata ?

Reply:
Secondary constructor moves the transformation logic to this file instead of having to be at the call site.